metadataproxy/log.py

```diff
@@ -0,0 +1,170 @@
+            'stream': 'ext://sys.stderr'
+        },
+    },
+    'loggers': {
+        'sqlalchemy.engine': {
+            'level': os.getenv('SQLALCHEMY_ENGINE_LOGGER_LEVEL' 'INFO'),
+            'propagate': True,
+        },
+        'werkzeug': {
+            'handlers': ['console_werkzeug'],
+            'level': 'DEBUG',
+            'propagate': False,
+        },
+    },
+    'root': {
+        'level': 'NOTSET',
+        'handlers': ['console', 'error']
+    },
+}
+
+normal_formatter = logging.Formatter(
+    fmt=LOG_SETTINGS['formatters']['normal']['format'],
+    datefmt=LOG_SETTINGS['formatters']['normal']['datefmt'])
+
+werkzeug_formatter = logging.Formatter(
+    fmt=LOG_SETTINGS['formatters']['werkzeug']['format'],
+    datefmt=LOG_SETTINGS['formatters']['werkzeug']['datefmt'])
+
+werkzeug_stream_handler = logging.StreamHandler(sys.stdout)
+werkzeug_stream_handler.setFormatter(werkzeug_formatter)
+
+werkzeug_error_stream_handler = logging.StreamHandler(sys.stderr)
+werkzeug_error_stream_handler.setFormatter(werkzeug_formatter)
+werkzeug_error_stream_handler.setLevel(logging.ERROR)
+
+stream_handler = logging.StreamHandler(sys.stdout)
+stream_handler.setFormatter(normal_formatter)
+
+error_stream_handler = logging.StreamHandler(sys.stderr)
+error_stream_handler.setFormatter(normal_formatter)
+error_stream_handler.setLevel(logging.ERROR)
+
+# environment flags we should honor
+BUGSNAG_API_KEY = os.getenv('BUGSNAG_API_KEY', None)
+SENTRY_DSN = os.getenv('SENTRY_DSN', None)
+
+logging.config.dictConfig(LOG_SETTINGS)
+
+if BUGSNAG_API_KEY:
+    import bugsnag
+
+if SENTRY_DSN:
+    from raven.conf import setup_logging
+    from raven.handlers.logging import SentryHandler
+    assert setup_logging
+
+
+def _configure_error_handler():
+    handler = None
+    if BUGSNAG_API_KEY:
+        client = bugsnag.Client(api_key=BUGSNAG_API_KEY)
+        handler = client.log_handler()
+        handler.setLevel(logging.ERROR)
+
+    if SENTRY_DSN:
+        handler = SentryHandler(SENTRY_DSN)
+        handler.setLevel(logging.ERROR)
+    return handler
+
+
+def get_logger(name, loglevel=None):
+    """
+    returns a python logger object configured according to seatgeek's standards
+
+    Args:
+        loglevel: a log level
+    """
+    logger = logging.getLogger(name)
+
+    # just in case that was not a new logger, get rid of all the handlers
+    # already attached to it.
+    del logger.handlers[:]
+
+    if not loglevel:
+        loglevel = logging.DEBUG
+
+    logger.setLevel(loglevel)
+    if name == 'werkzeug':
+        logger.addHandler(werkzeug_stream_handler)
+        logger.addHandler(werkzeug_error_stream_handler)
+    else:
+        logger.addHandler(stream_handler)
+        logger.addHandler(error_stream_handler)
+
+    logger.propagate = False
+    error_handler = _configure_error_handler()
+    if error_handler:
+        logger.addHandler(error_handler)
+
+    return logger
+
+
+def log_request(request, response, logger=None):
```

Comment:
This should be configurable through a gunicorn log handler, which will automatically handle this on requests. The nice thing about letting gunicorn's log handlers handle this is that people can choose their own logging format, rather than this, which is hardcoded.